lib/store_web/live/cart_live/index.ex

```diff
@@ -54,14 +53,30 @@ defmodule StoreWeb.CartLive.Index do
       |> Repo.one()
       |> Repo.preload(:products)
 
-    order_product =
-      OrdersProducts
-      |> where(order_id: ^order.id)
-      |> where(product_id: ^id)
-      |> Repo.one()
+    quantity = Enum.count(order.products)
+
+    if quantity == 1 do
+      order
+      |> Order.changeset(%{status: :canceled})
+      |> Repo.update!()
```

Comment:
This should also be refactored to its context. Create a `update_order_status/1` function there, for example.

Reply:
My bad I have that function, I should have done that since the beginning.